packages/backend/src/telemetry-notification.ts

```diff
@@ -0,0 +1,80 @@
+import fs from 'node:fs/promises';
+import os from 'node:os';
+import path from 'node:path';
+
+// if we end up storing anything else in this file, will need to extract this type
+// however, this is the only place it's used at the moment.
+interface ClerkConfigFile {
+  telemetryNoticeVersion: string;
+}
+
+// If we make significant changes to how telemetry is collected in the future, bump this version.
+const TELEMETRY_NOTICE_VERSION = '1';
+
+function telemetryNotice() {
+  console.log('╭────────────────────────────────────────────────────────────────╮');
+  console.log('│ Attention: Clerk now collects telemetry data from its          │');
+  console.log('│ SDKs when connected to development instances.                  │');
+  console.log('│                                                                │');
+  console.log("│ The data collected is used to inform Clerk's product roadmap.  │");
+  console.log('│ To learn more, including how to opt-out from telemetry, visit: │');
+  console.log('│                                                                │');
+  console.log('│ https://beta.clerk.com/docs/telemetry.                         │');
+  console.log('╰────────────────────────────────────────────────────────────────╯');
+}
+
+// Adapted from https://github.com/sindresorhus/env-paths
+function getConfigDir(name: string) {
+  const homedir = os.homedir();
+  const macos = () => path.join(homedir, 'Library', 'Preferences', name);
+  const win = () => {
+    // eslint-disable-next-line turbo/no-undeclared-env-vars
+    const { APPDATA = path.join(homedir, 'AppData', 'Roaming') } = process.env;
+    return path.join(APPDATA, name, 'Config');
+  };
+  const linux = () => {
+    // eslint-disable-next-line turbo/no-undeclared-env-vars
+    const { XDG_CONFIG_HOME = path.join(homedir, '.config') } = process.env;
+    return path.join(XDG_CONFIG_HOME, name);
+  };
+  switch (process.platform) {
+    case 'darwin':
+      return macos();
+    case 'win32':
+      return win();
+    default:
+      return linux();
+  }
+}
+
+async function notifyAboutTelemetry() {
```

Comment:
@clerk/backend is an isomorphic package -  have we tested that this works across all runtimes, including CF workers, netlify etc?

Reply:
And a quick follow up: what's the plan for telemetry on the clerk-js side?

Reply:
I have not - just tested locally. Great catch! I'm not sure how we normally test into those runtimes, do you have any links to things I could read to learn about how to do so? I assume the main thing that could break down is the persistence of the config file here.